ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -153,8 +133,11 @@ public HelixClusterManager(ClusterMapConfig clusterMapConfig, String instanceNam
               manager.connect();
               logger.info("Established connection to Helix manager at {}", zkConnectStr);
             }
-
-            ClusterChangeHandler clusterChangeHandler = new ClusterChangeHandler(dcName);
+            ClusterChangeHandler clusterChangeHandler =
+                new SimpleClusterChangeHandler(clusterMapConfig, dcName, selfInstanceName, partitionOverrideInfoMap,
```

Comment:
When we add a more dynamic ClusterChangeHandler implementation in a future PR, could we look into ways to avoid passing these shared maps directly into the ClusterChangeHandler impls? This could be accomplished by providing an `addPartitionIfAbsent` callback that would internally handle synchronization. Currently, it is up to the impl to correctly handle synchronization across multiple mutable data structures.

Reply:
Good point, will keep it in mind when implementing dynamic clusterChangeHandler.